File: rb/features.rb
# DomainWhois SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DomainWhoisFeatures
  def self.make_feature(name)
    case name
    when "base"
      DomainWhoisBaseFeature.new
    when "test"
      DomainWhoisTestFeature.new
    else
      DomainWhoisBaseFeature.new
    end
  end
end

File: py/features.py
# DomainWhois SDK feature factory

from feature.base_feature import DomainWhoisBaseFeature
from feature.test_feature import DomainWhoisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DomainWhoisBaseFeature(),
        "test": lambda: DomainWhoisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
